### ISSUE TITLE
passing validSuffix not working as expected

### ISSUE THREAD

**Issue by davidbielik** [remusao/tldts#512]
Hello again!

I'm trying to add a few other domains to be treated as valid suffixes, using the `validHosts` API option.

Am I using this incorrectly? or is it possible to add my own list of public suffixes? If not I have a workaround I can use outside of this library 😃 

```
const {parse} = require('tldts');
const parsed = parse("subdomain.wordpress.com", {
   allowPrivateDomains: true,
   validHosts: ['wordpress.com']
});
expect(parsed.domain).to.equal("sub.wordpress.com");
// domain actually equals "wordpress.com"
```

![image](https://user-images.githubusercontent.com/7410770/78792657-69241000-797f-11ea-8bf2-2b6759b6f6f4.png)

**Comment by remusao** [comment]
Hi again @davidbielik,

I am afraid that at the moment there is no way to extend the list of public suffixes dynamically. This was a choice I made early on as my top-priority was speed and I sacrificed a bit of flexibility for this. As a result, `tldts` will have built-in lists which are compiled in a very efficient form at build-time but cannot be changed at runtime.

In principle this could be changed, as long as I can do it without impacting performance of other methods.

I'm sorry to not have a more satisfying answer to give you at this point!
Best,
